handlers/mapping_handlers.go

```diff
@@ -0,0 +1,404 @@
+		return
+	}
+
+	sessionMember, err := env.DB.GetUserConversationMapping(userID, conversationID)
+	if err != nil {
+		internalServerError(w, err)
+		return
+	}
+	if sessionMember == nil {
+		errMsg := fmt.Sprintf("User %d is not in conversation %d", userID, conversationID)
+		log.Println(errMsg)
+		http.Error(w, "Conversation not found", http.StatusNotFound)
+		return
+	}
+
+	members, err := env.DB.GetUserConversationMappings(conversationID)
+	if err != nil {
+		internalServerError(w, err)
+		return
+	}
+	memberList := &models.UserConversationMappingList{Users: members}
+
+	// TODO: Get user names from karen?
+
+	w.Header().Add("Content-Type", "application/json")
+	json.NewEncoder(w).Encode(memberList)
+}
+
+// PatchMappingHandler updates a single user in a conversation
+func (env *Env) PatchMappingHandler(w http.ResponseWriter, r *http.Request) {
+	defer r.Body.Close()
+	userID, err := strconv.ParseInt(r.Header.Get("User-ID"), 10, 64)
+	if err != nil {
+		errMsg := "Invalid user ID"
+		log.Println(errMsg + ": " + err.Error())
+		http.Error(w, errMsg, http.StatusBadRequest)
+		return
+	}
+
+	vars := mux.Vars(r)
+	conversationID, err := strconv.ParseInt(vars["conversation_id"], 10, 64)
+	if err != nil {
+		errMsg := "Invalid conversation ID"
+		log.Println(errMsg + ": " + err.Error())
+		http.Error(w, errMsg, http.StatusBadRequest)
+		return
+	}
+	memberID, err := strconv.ParseInt(vars["user_id"], 10, 64)
+	if err != nil {
+		errMsg := "Invalid user ID"
+		log.Println(errMsg + ": " + err.Error())
+		http.Error(w, errMsg, http.StatusBadRequest)
+		return
+	}
+	if userID == memberID {
+		errMsg := "Cannot modify own role"
+		log.Println(errMsg)
+		http.Error(w, errMsg, http.StatusForbidden)
+		return
+	}
+
+	conversation, err := env.getConversation(w, conversationID)
+	if err != nil || conversation == nil {
+		return
+	}
+
+	sessionMember, err := env.DB.GetUserConversationMapping(userID, conversationID)
+	if err != nil {
+		internalServerError(w, err)
+		return
+	}
+	if sessionMember == nil {
+		errMsg := fmt.Sprintf("User %d is not in conversation %d", userID, conversationID)
+		log.Println(errMsg)
+		http.Error(w, "Conversation not found", http.StatusNotFound)
+		return
+	}
+
+	member, err := env.DB.GetUserConversationMapping(memberID, conversationID)
+	if err != nil {
+		internalServerError(w, err)
+		return
+	}
+	if member == nil {
+		errMsg := fmt.Sprintf("User %d is not in conversation %d", userID, conversationID)
+		log.Println(errMsg)
+		http.Error(w, "User not found", http.StatusNotFound)
+		return
+	}
+
+	reqMember := &models.UserConversationMapping{}
+	if err := parseMappingJSON(w, r.Body, reqMember); err != nil {
+		return
+	}
+
+	if reqMember.Role == "" && reqMember.Nickname == nil {
+		errMsg := "Request body is missing field(s)"
+		log.Println(errMsg)
+		http.Error(w, errMsg, http.StatusBadRequest)
+		return
+	}
+
+	if reqMember.Role != "" {
+		if reqMember.Role != models.Admin && reqMember.Role != models.User {
```

Comment:
Might help with readability if you use the `Valid` method for the `Role` type and then the `reqMember.Role != models.Owner`.